app/src/main/java/com/mikhailovskii/weatherandroid/ui/choose_location/ChooseLocationPresenter.kt

```diff
@@ -0,0 +1,42 @@
+package com.mikhailovskii.weatherandroid.ui.choose_location
+
+import com.google.firebase.firestore.FirebaseFirestore
+import com.mikhailovskii.weatherandroid.data.entities.User
+import com.mikhailovskii.weatherandroid.ui.base.BasePresenter
+import com.mikhailovskii.weatherandroid.util.Preference
+
+class ChooseLocationPresenter : ChooseLocationContract.ChooseLocationPresenter,
+    BasePresenter<ChooseLocationContract.ChooseLocationView>() {
+
+    private var database = FirebaseFirestore.getInstance()
+
+    override fun saveLocation(location: String) {
+        val user = Preference.user
+        user?.location = location
+        Preference.user = user
+
+        saveLocationInFirebase(user)
+
+        view?.onLocationSaved()
+    }
+
+    private fun saveLocationInFirebase(user: User?) {
+        database.collection(USERS_COLLECTION).whereEqualTo(LOGIN_FIELD, user?.login).get()
+            .addOnSuccessListener { databaseResult ->
+
+                val map: Map<String, String> = hashMapOf(LOCATION to (user?.location ?: ""))
+
+                for (document in databaseResult) {
+                    database.collection(USERS_COLLECTION).document(document.id).update(map)
+                }
+
+            }
+    }
+
+    companion object {
+        private const val USERS_COLLECTION = "users"
```

Comment:
@SergeiMikhailovskii I see you use the same Firebase api in two presenters now. I guess it's time to think of some kind of the next level of abstraction like `FirebaseRepository` that will handle all the saving/reading operations to Firebase (make an interface and try not to be Firebase-specific in designing the API of the repository, so you could easily replace it with a local database (or amazon or anything else) repository if you need some day). The repository could be injected into both presenter and use its API to save/read the Firebase storage. 

**Please note that this is not required to be done in scope of this pull request, but you may create a new task to implement it later**